### ISSUE TITLE
Consider using deck.gl instead of mapbox/maplibre

### ISSUE THREAD

**Issue by kaligrafy** [chairemobilite/transition#552]
Deck.gl is more mature than it was when transition development started. It now support everything we need and seems more light and easy to use for events than mapbox/maplibre. It also accept multiple maps API as base layers (mapbox, google maps, arcgis, etc.). It support webGL 2 out of the box and can work with only simple tiles instead of loading an entire library for base layers.
[Deck.gl](https://deck.gl/)
License is MIT

**Comment by tahini** [comment]
TODO for this feature, @kaligrafy, see if I'm forgetting something for the initial port to be fully complete. It may sound easy, but each of those elements needs migration work.

- [ ] Show enabled layers per section in a deck.gl canvas
- [ ] Make sure they can be interactively updated (when adding, deleting paths/nodes)
- [ ] Make events work on features (lines and nodes)
- [ ] Make events work on maps (not specific to a layer)
- [ ] Make sure all popups work and are at their place
- [ ] Make sure the map is navigable and properties are kept in the user's preferences
- [ ] Allow to hide/show specific features of a layer (with the visibility toggle)
- [ ] Use custom shaders for the selected line/nodes (may not be exactly the same as currently, but it will keep the idea)
- [ ] Add a map canvas, either using maplibre or mapbox
- [ ] Support custom tiles URL for background
- [ ] Support the polygon draw selection tool (see if it can be generalized right now, instead of just being happy with selecting nodes)
- [ ] chaire-lib should have no dependency to mapbox anymore

**Comment by kaligrafy** [comment]
Seems good for me, I don't see anything else!

The map canvas you mean the vector road network?

**Comment by tahini** [comment]
> Seems good for me, I don't see anything else!
> 
> The map canvas you mean the vector road network?

Yes I mean the current mapbox background or maplibre (deck.gl itself is just layers with no visible data)

**Comment by kaligrafy** [comment]
yes, that is correct and in the long run we could generate our own vector tiles

**Comment by davidmurray** [comment]
@tahini let me know if you need some help for this!
